buildspec.yml

```diff
@@ -3,61 +3,106 @@ version: 0.2
 env:
   variables:
     FRAMEWORK_FULL_VERSION: '1.4.1'
-    FRAMEWORK_SHORT_VERSION: '1.4'
-    AWS_DEFAULT_REGION: 'us-west-2'
-    ECR_REPO: 'sagemaker-mxnet-serving'
+    CPU_PY_VERSION: '2'
+    CPU_INSTANCE_TYPE: 'ml.c4.xlarge'
+    GPU_PY_VERSION: '3'
+    GPU_INSTANCE_TYPE: 'ml.p2.xlarge'
+    ECR_REPO: 'sagemaker-test'
     GITHUB_REPO: 'sagemaker-mxnet-serving-container'
-    EI_ACCELERATOR_TYPE: 'ml.eia1.medium'
-    GPU_INSTANCE_TYPE: 'p2.xlarge'
     SETUP_FILE: 'setup_cmds.sh'
     SETUP_CMDS: '#!/bin/bash\npip install --upgrade pip\npip install -U -e .\npip install -U -e .[test]'
 
 phases:
   pre_build:
     commands:
       - start-dockerd
-
       - ACCOUNT=$(aws sts get-caller-identity --query 'Account' --output text)
+      - PREPROD_IMAGE="$ACCOUNT.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$ECR_REPO"
+      - PR_NUM=$(echo $CODEBUILD_SOURCE_VERSION | grep -o '[0-9]\+')
+      - echo 'Pull request number:' $PR_NUM '. No value means this build is not from pull request.'
 
   build:
     commands:
+      # install
+      - pip3 install -U -e .
+      - pip3 install -U -e .[test]
+
       # run linter
       - tox -e flake8
 
       # run unit tests
-      - tox -e py27,py36 test/unit
+      - pytest test/unit
 
-      # create pip archive (tar name will be something like sagemaker_mxnet_serving_container-1.0.0.tar.gz, but the Dockerfiles expect sagemaker_mxnet_serving_container.tar.gz)
+      # Create pip archive in dist/
       - python3 setup.py sdist
       - mv dist/sagemaker_mxnet_serving_container-*.tar.gz dist/sagemaker_mxnet_serving_container.tar.gz
       - cp dist/sagemaker_mxnet_serving_container.tar.gz docker/$FRAMEWORK_FULL_VERSION/py3/sagemaker_mxnet_serving_container.tar.gz
 
-      # build images
-      - python3 scripts/build_all.py --version $FRAMEWORK_FULL_VERSION --account $ACCOUNT --repo $ECR_REPO
+      - build_dir="docker/$FRAMEWORK_FULL_VERSION"
+      - build_id="$(echo $CODEBUILD_BUILD_ID | sed -e 's/:/-/g')"
 
-      # run cpu local integration tests
-      - IGNORE_COVERAGE=- tox -e py36 -- test/integration/local --py-version 2,3 --processor cpu --framework-version $FRAMEWORK_FULL_VERSION --region $AWS_DEFAULT_REGION --docker-base-name $ECR_REPO
+      # build cpu image
+      - cpu_dockerfile="Dockerfile.cpu"
+      - CPU_TAG="$FRAMEWORK_FULL_VERSION-cpu-py$CPU_PY_VERSION-$build_id"
 
-      # push docker images to ECR
-      - python3 scripts/publish_all.py --version $FRAMEWORK_FULL_VERSION --account $ACCOUNT --repo $ECR_REPO
+      - docker build -f $build_dir/py$CPU_PY_VERSION/$cpu_dockerfile --build-arg framework_support_installable=$tar_name -t $PREPROD_IMAGE:$CPU_TAG .
```

Comment:
I think it'd be better to use the same building scripts and test commands as the release build, and just modify `build_all.py` and `publish_all.py` to take the build ID as an argument, and then add something like:

```python
if args.build_id:
    tag += '-{}'.format(args.build_id)
```

Reply:
Added `--build-id` option in build_all.py and publish_all.py